misk/src/main/kotlin/misk/web/WebConfig.kt

```diff
@@ -101,3 +104,30 @@ data class WebUnixDomainSocketConfig(
   /** If true, the listener will support H2C. */
   val h2c: Boolean? = true
 )
+
+data class CorsConfig(
+    /** A comma separated list of origins that are allowed to access the resources. */
+    val allowedOrigins: Array<String> = arrayOf("*"),
+    /**
+     * A comma separated list of HTTP methods that are allowed to be used when
+     * accessing the resources.
+     */
+    val allowedMethods: Array<String> = arrayOf("GET", "POST", "HEAD"),
+    /**
+     * A comma separated list of HTTP headers that are allowed to be specified when
+     * accessing the resources.
+     */
+    val allowedHeaders: Array<String> = arrayOf("X-Requested-With", "Content-Type", "Accept",
+        "Origin"),
+    /** A boolean indicating if the resource allows requests with credentials. */
+    val allowCredentials: Boolean = true,
+    /** The number of seconds that preflight requests can be cached by the client. */
+    val preflightMaxAge: String = "1800",
+    /**
+     * If true preflight requests are chained to their target resource for normal handling
```

Comment:
supernit: I think this might read better as “True if preflight requests..." rather than “If true preflight requests”